src/EditorFeatures/CSharpTest/AddParameter/AddParameterTests.cs

```diff
@@ -584,5 +584,1027 @@ void M()
+        {
+            var code =
+@"
+    class C1
+    {
+        void M1() { }
+        void M2()
+        {
+            [|M1|]<bool>(true);
+        }
+    }
+";
+            var fix0 =
+@"
+    class C1
+    {
+        void M1<T>(T v) { }
+        void M2()
+        {
+            M1<bool>(true);
+        }
+    }
+";
+            await TestInRegularAndScriptAsync(code, fix0);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocation_ExisitingTypeArgumentIsNotGeneralized()
+        {
+            var code =
+@"
+    class C1
+    {
+        void M1<T>(T v) { }
+        void M2()
+        {
+            [|M1|](true, true);
+        }
+    }
+";
+            var fix0 =
+@"
+    class C1
+    {
+        void M1<T>(T v, bool v1) { }
+        void M2()
+        {
+            M1(true, true);
+        }
+    }
+";
+            await TestInRegularAndScriptAsync(code, fix0);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationPartialClasses()
+        {
+            var code =
+@"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" CommonReferences=""true"">
+        <Document>
+namespace N1
+{
+    partial class C1
+    {
+        partial void PartialM();
+    }
+}
+        </Document>
+        <Document>
+namespace N1
+{
+    partial class C1
+    {
+        partial void PartialM() { }
+        void M1()
+        {
+            [|PartialM|](1);
+        }
+    }
+}
+        </Document>
+    </Project>
+</Workspace>";
+            var fix0 =
+@"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" CommonReferences=""true"">
+        <Document>
+namespace N1
+{
+    partial class C1
+    {
+        partial void PartialM(int v);
+    }
+}
+        </Document>
+        <Document>
+namespace N1
+{
+    partial class C1
+    {
+        partial void PartialM(int v) { }
+        void M1()
+        {
+            PartialM(1);
+        }
+    }
+}
+        </Document>
+    </Project>
+</Workspace>";
+            await TestInRegularAndScriptAsync(code, fix0);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        [Trait("TODO", "Code formatting")]
+        public async Task TestObjectCreationAddTypeParameter_AddTypeParameterIfUserSpecifiesOne_TypeParameterAndParameterArgument()
+        {
+            var code =
+@"
+    class C1
+    {
+        C1() { }
+        void M2()
+        {
+            var c = new [|C1<bool>|](true);
+        }
+    }
+";
+            var fix0 =
+@"
+    class C1<T>
```

Comment:
making a type generic is pretty heavyweight.  not sure we want the feature to do this.